### PR TITLE
[PWA] Feat/263 coupon codes

### LINKED ISSUE #263: Share-able links that apply coupon codes
- [x] Enable share-able links that apply coupon codes
- [x] Logic on every page that looks for `y1` query parameters and stores them in a cookie.
- [x] Cart &
- [x]  checkout can read the cookie, apply the coupon and delete the cookie.
- [x] Include `d1` logic as per https://stage.maidenform.com/static/frontend/Magento/base/default/Born_MarketingChannel/js/marketingchannel.min.js
- [ ] Track `d1` value in ACDL.

### PR DESCRIPTION
**Pull request by dicagno**
### What is missing?
- Handling d1 parameter with ACDL (awaiting docs)

Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #263 

Test URLs:
- https://feat-263-coupon-codes--maidenform--hlxsites.hlx.live/

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -389,10 +389,58 @@ export async function getFragmentFromFile(fileURL) {
...
+
+function detectAndApplyCoupon() {
+  const couponParam = getCouponQueryParam();
+  if (couponParam && couponParam.length > 3) {
+    console.log('couponParam', couponParam);
+    applyCouponAsCookie(couponParam);
+  }
+}
+
+function performCommeceChecks() {
```
Comment (fe-lix-) [inline scripts/scripts.js:435]: Typo `performCommerceChecks`

**Review by fe-lix- — COMMENTED** [review]
LGTM

**Comment by hannessolo** [comment]
@dicagno feel free to open a follow up issue for ACDL and merge this - Neela will send us the code snippet for acdl in the next days.

**Comment by dicagno** [comment]
@hannessolo docs ready for review, should be sufficient for my task: https://github.com/adobe/commerce-events/blob/ef01ed6d3b812680bd26b69bc271c8e17e2821bd/examples/events/custom.md

**Comment by hannessolo** [comment]
@dicagno yes, I saw the docs - feel free to add an event. We can discuss with Neela to validate that the approach will also work in the backend in the sync today. I can forward the meeting to you.

**Review by hannessolo — APPROVED** [review]

**Pull request merged**